operator/internal/controller/transactioninvoke_controller.go

```diff
@@ -212,12 +214,38 @@ func getContractDeploymentAddress(ctx context.Context, c client.Client, name, na
 
 }
 
+// TODO this is duplicated btwn here and SmartContractDeploymentReconciler, as reconcileAll alludes to, generics
```

Comment:
We have some helpers to allow generics. Found it surprisingly hard to support them in k8s data structures, I had to introduce a thing like this to every file:
https://github.com/LF-Decentralized-Trust-labs/paladin/blob/ef6a417de44129e9622934d3306bd1b72fe55d28/operator/internal/controller/transactioninvoke_controller.go#L49-L56